tools/proto-annotations/cmd/etcd_version.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2021 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/coreos/go-semver/semver"
+	"go.etcd.io/etcd/server/v3/storage/wal"
+
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+)
+
+// printEtcdVersion writes etcd_version proto annotation to stdout and returns any errors encountered when reading annotation.
+func printEtcdVersion() []error {
+	var errs []error
+	annotations, err := allEtcdVersionAnnotations()
+	if err != nil {
+		errs = append(errs, err)
+		return errs
+	}
+	sort.Slice(annotations, func(i, j int) bool {
+		return annotations[i].fullName < annotations[j].fullName
+	})
+	output := &bytes.Buffer{}
+	for _, a := range annotations {
+		newErrs := a.Validate()
+		if len(newErrs) == 0 {
+			err := a.PrintLine(output)
+			if err != nil {
+				errs = append(errs, err)
+				return errs
+			}
+		}
+		errs = append(errs, newErrs...)
+	}
+	if len(errs) == 0 {
+		fmt.Print(output)
+	}
+	return errs
+}
+
+func allEtcdVersionAnnotations() (annotations []etcdVersionAnnotation, err error) {
+	var fileAnnotations []etcdVersionAnnotation
+	protoregistry.GlobalFiles.RangeFiles(func(file protoreflect.FileDescriptor) bool {
+		switch string(file.Package()) {
+		// Skip external packages that are not expected to have etcd version annotation.
+		case "io.prometheus.client", "grpc.binarylog.v1", "google.protobuf", "google.rpc", "google.api":
```

Comment:
nit: please move it to the top level constant.

Reply:
Done